src/components/SortButton.tsx

```diff
@@ -0,0 +1,176 @@
+    onSortChange(sortOrder)
+    setDropdownOpen(false)
+  }
+
+  return (
+    <button
+      onClick={() => setDropdownOpen(!isDropdownOpen)}
+      className={css({
+        width: '45px',
+        border: 'solid 1px {colors.gray50}',
+        backgroundColor: 'darkgray',
+        borderRadius: '8px',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'relative',
+      })}
+    >
+      <div
+        className={css({
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        })}
+      >
+        <SortIcon size={20} fill={isLightTheme ? '{colors.lightgray}' : '{colors.fg}'} />
+      </div>
+      <div
+        className={cx(
+          css({
+            opacity: 0,
+            visibility: 'hidden',
+            transition: 'opacity 0.3s ease-in-out, visibility 0.3s ease-in-out',
+            willChange: 'opacity, visibility',
+            position: 'absolute',
+            top: '100%',
+            marginTop: '0.5rem',
+            right: 0,
+            backgroundColor: 'bg',
+            border: 'solid 1px {colors.gray50}',
+            borderRadius: '8px',
+            zIndex: 'dialog',
+            padding: '1rem',
+            display: 'flex',
+            flexDirection: 'column',
+            gap: '0.5rem',
+          }),
+          isDropdownOpen &&
+            css({
+              opacity: 1,
+              visibility: 'visible',
+            }),
+        )}
+      >
+        <h2
+          className={css({
+            color: 'gray50',
+            borderBottom: 'none',
+            textAlign: 'left',
+            fontSize: '0.9rem',
+            margin: '0',
+          })}
+        >
+          Sort by:
+        </h2>
+        <label
+          className={css({
+            display: 'flex',
+            flexDirection: 'row',
+            color: 'gray50',
+          })}
+        >
+          <input
+            type='radio'
+            checked={selectedSort === 'alphabetical'}
+            onChange={() => handleSortChange('alphabetical')}
+            className={css({
+              appearance: 'none',
+              width: '12px',
+              height: '12px',
+              border: '2px solid {colors.gray50}',
+              borderRadius: '50%',
+              display: 'inline-block',
+              position: 'relative',
+              '&:checked': {
+                borderColor: 'fg',
+              },
+            })}
+          />
+          <h3
+            className={css({
+              color: selectedSort === 'alphabetical' ? '{colors.fg}' : '{colors.gray50}',
+              margin: '0 0 0 0.2rem',
+              fontWeight: 'normal',
+              fontSize: '0.9rem',
+            })}
+          >
+            Alphabetical
+          </h3>
+        </label>
+        <label
+          className={css({
+            display: 'flex',
+            flexDirection: 'row',
+          })}
+        >
+          <input
```

Comment:
I think there is enough duplicated code here to warrant factoring it out into a `SortOption` component. It can encapsulate the label + input + h3.